internal/provider/vsphere/inventory.go

```diff
@@ -0,0 +1,49 @@
+package vsphere
+
+import (
+	"github.com/openshift/assisted-service/internal/usage"
+	"github.com/openshift/assisted-service/models"
+)
+
+func (p *vsphereProvider) CleanPlatformValuesFromDBUpdates(updates map[string]interface{}) error {
+	updates[DbFieldVsphere] = nil
```

Comment:
As mentioned before, "DbFieldVsphere" is wrong and probably not needed. However, If we reset the vsphere database fields we might need to update "DbFieldPlatformType" to the default value, which should be "models.PlatformTypeBaremetal"

Reply:
This is why its best to save it for a follow up PR as commented, all that discussion should be made on a dedicated PR for your proposed change